### PR TITLE
swc university course draft with edits moved from site repo

### PR DESCRIPTION
**Pull request by chendaniely**
original PR was sent from: https://github.com/chendaniely/site/blob/post/swdcretreat-university_course_roundtable/blog/2015/11/swdcretreat2015-university_course.html

The original PR from the old site repo: https://github.com/swcarpentry/site/pull/1173

### CODE REVIEW

**Comment by chendaniely** [comment]
@billMills @ttimbers @jennybc

I moved the post to the new website repo and made some more edits

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+<a href='https://twitter.com/JennyBryan'>Jenny Bryan</a>.
+You can <a href='https://plus.google.com/events/c41eaub933br0odpb385rcg9j00'>watch</a> the discussion and/or
+<a href='http://pad.software-carpentry.org/swc-instructor-retreat-2015-7PM-UTC'>read</a> the notes on etherpad.
+
+<!-- end excerpt -->
+
+First, thanks to everyone who joined the discussion:
+<ul>
+  <li>Tiffany Timbers / @TiffanyTimbers</li>
+  <li>Jenny Bryan / @JennyBryan </li>
```
Comment (jennybc) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: Are these meant to be twitter handles (in which we're good) or github usernames (in which case I'm @jennybc)?

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+  </h4>
+</p>
+
+<p>
+  [J] If someone at UBC wanted to offer this, and isn't me (with teaching something or with rights),
+  I would suggest to teach it as a variable special topics.
+  The other option is to find a course that is related, and offer to teach a lap with them.
+  You can sell the SWC stuff as a lab to an an existing course.
+  This would also provide the added benefit of getting you paid as a TA, for example.
+</p>
```
Comment (jennybc) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html:117]: Pls replace with: Two good options for introducing SWC-type material: For faculty or others who teach regular courses, one could offer it as a special topics course. For grad students who normally TA, one could offer to run a hands-on computing lab for a related, existing course.

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+  Since code review was part of the course, learning to send PRs is essential.
+  These sessions were split up a week apart.
+  Then we covered all the Python basics.
+  Data munging and visualizations are really important, so we spend a lot of time with Visualizations and Pandas.
+  Because I think you learn a lot by doing, almost the last 9 hours of the course are done doing hack sessions, where I am around to help.
+  Quest also prefers courses not to be lecture based to promote active learning, so I'm not teaching during the hack sessions.
+</p>
+
+<p>
+  [J] I'm not teaching classic SWC material
```
Comment (jennybc) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: Pls replace the sentences prior to "One of ..." with: I don't necessarily teach classic SWC material in the same format used in two-day workshops. I incorporate it as we go through an existing course.

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+    Tiffany, can you talk about the <a href='https://phy3009.github.io/PHY3009-2015/'>course</a> you've been beta testing?
+  </h4>
+</p>
+
+<p>
+  [T]: Anthony has been teaching the course to undergraduate students and I've been building off Katy and Anthony's experiences from teaching as well as my own from teaching Software Carpentry.
+  I decided to use this textbook as my course at Quest.
+  The book is based on Python, but many of the principles are language agnostic.
+  The book covers many topics SWC teaches, such as Bash, Python, Make, documentation, licenses, and many open science aspects.
+  Katy and Anthony also created a <a href='https://github.com/physics-codes/examples'>Github repo</a> of annotated iPython (Jupyter) notebooks of the code in the book, as well as an <a href='https://github.com/physics-codes/seminar'>example syllabus</a>.
```
Comment (ttimbers) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: please change the url here to: https://github.com/physics-codes/seminar, and remove ", as well as an <a href='https://github.com/physics-codes/seminar'>example syllabus</a>" as there is no syllabus.

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+</p>
+
+<p>
+  [T] My current course is taught as a 3.5 week course.
+  This is supposed to be the same number of hours as a full semester course.
+  We cover an abbreviated shell and 6 hours of version control.
+  Since code review was part of the course, learning to send PRs is essential.
+  These sessions were split up a week apart.
+  Then we covered all the Python basics.
+  Data munging and visualizations are really important, so we spend a lot of time with Visualizations and Pandas.
```
Comment (ttimbers) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: please change "so we spend" to "so we spent"

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html [hunk 6]
```diff
@@ -0,0 +1,437 @@
...
+  I've had people from 50 different programs.  25 of which have at least 2 students.
+  So basically, they are all over.
+  40% are cumulatively statistics since I have all the incoming masters students.
+  Computational biology and bioinformatics are usually the others taking the course.
+  I think it's good to have this diversity in the class and not be hyper specific.
+</p>
+
+<p>
+  [T] I have been using Katy and Anthony's book for this course because I was hired by the department of physical science at Quest.
+  Dispite 'physics' being in the book and course title, I have a wide variety of people who take my class.
```
Comment (ttimbers) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: Please change "Dispite" to "Despite"

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html [hunk 7]
```diff
@@ -0,0 +1,437 @@
...
+
+<p>
+  [T] We do 3 hours of shell the first day.
+  3 Hours of git the next.
+  I think it depends how many active learning exercise you can get.
+  For me it took me 9 Hours to get through loops, functions, and if statements because I had a lot of active learning exercise because I wanted them to fully understand these concepts.
+  Testing took another 3 hours (in the third week of the course).
+  Essentially I do 2.5 weeks of SWC then we do code review.
+  Each student is given the other student's homework to review.  And I sit with them and they give me a summary of the pull request.
+  At the same time the rest of the class are working on their final project.
```
Comment (ttimbers) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: Please change "class **are** working" to "class **is** working"

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html [hunk 8]
```diff
@@ -0,0 +1,437 @@
...
+    What final projects do you have the students work on?  Do you assign them a project or do they bring their own questions?
+  </h4>
+</p>
+
+<p>
+  [T] I've been extremely flexible with this.
+  I have them think about the final project from Day 1.
+  They pick something in the area they are interested in and have a research question with a hypothesis and do something that requires a computer to do.
+  These are undergraduates, so my expectations are not as high because they do not have as much research experience.
+  They must have a Python script import data and save a plot or table from an analysis or visualization
```
Comment (ttimbers) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: Please change "They must have" to "The project end goal is to have"

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html [hunk 9]
```diff
@@ -0,0 +1,437 @@
...
+  and have a record in either a Makefile or Shell script to make sure everything is reproducible.
```
Comment (ttimbers) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: Please change "have a record in" to "have a record of the Python script inputs in"

Comment (ttimbers) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: Please change "have a record in" to "have a record of the Python script inputs in"

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html [hunk 10]
```diff
@@ -0,0 +1,437 @@
...
+
+<p>
+  [J] An <em>army</em> of TAs is critical.  I have 6-7 (not all full time).
+  The department here was able to prove that this is not just a statistics class anymore, and people are coming from all over campus.
+  So I got external funding for more TAs than the average graduate class.
+  I only have graduate TAs, no undergraduate TAs.
+</p>
+
+<p>
+  [T] Unfortunately Quest is a very new institute with limited funding.  So I am the only TA/Helper/Teacher/Office Hours.
```
Comment (ttimbers) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: I misspoke here, please change this response to: "During class I have no helpers or TAs, but outside of the class there are are peer helpers available."

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html [hunk 11]
```diff
@@ -0,0 +1,437 @@
...
+  I started by chasing the technology for years, but every year almost required a completely new course re-write.
+  I eventually generalized it to the key genomics pipelines and statistical analysis.
+  For example, multiple comparison and empirical bayes.
+  I try not to chase specific sequencing platforms.
+  You can only be chasing technology a few places at a time.
+  <br>
+  Research wise, I've only been able to put these classes after I've gotten tenure.
+  This allmws me to put more effort into developing the course with fewer outside distractions.
+  I would say what I'm doing would not mesh well with being super productive research-wise
+  ***Jeff Leek***, Roger Peng, rafa irizarry, are doing on line courses and super productive research-wise.
```
Comment (chendaniely) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: @jennybc can you confirm the spelling of names?

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html (hunk 1 above)
Comment (chendaniely) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: they're twitter handles

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html [hunk 12]
```diff
@@ -0,0 +1,437 @@
...
+  I started by chasing the technology for years, but every year almost required a completely new course re-write.
+  I eventually generalized it to the key genomics pipelines and statistical analysis.
+  For example, multiple comparison and empirical bayes.
+  I try not to chase specific sequencing platforms.
+  You can only be chasing technology a few places at a time.
+  <br>
+  Research wise, I've only been able to put these classes after I've gotten tenure.
+  This allmws me to put more effort into developing the course with fewer outside distractions.
+  I would say what I'm doing would not mesh well with being super productive research-wise
+  Jeff Leek, Roger Peng, rafa irizarry, are doing on line courses and super productive research-wise.
```
Comment (chendaniely) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: @jennybc Can you check to see if the names are spelled correctly?
I know I'm missing capitalization on Rafa

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html (hunk 1 above)
Comment (jennybc) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: Should you make them links then?

File: _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html (hunk 12 above)
Comment (jennybc) [inline _posts/2016/02/2016-02-03-swdcretreat2015-university_course.html]: Rafael Irizarry

**Comment by chendaniely** [comment]
@gvwilson I think this is ready, might need a last minute tweek with the dates and YAML before the final merge

**Comment by jduckles** [comment]
@chendaniely please make the date changes so its ready to go and I'll merge.

**Comment by chendaniely** [comment]
@jduckles changed dates in in f2a9e445c55e33e28a43846fa66eed0afdd8f128

**Pull request merged**